src/qibo/quantum_info/basis.py

```diff
@@ -100,28 +100,37 @@ def pauli_basis(
     else:
         basis_full = basis_single
 
+    basis_full = backend.cast(basis_full, dtype=basis_full[0].dtype)
+
     if vectorize and sparse:
         basis, indexes = [], []
         for row in basis_full:
             row = vectorization(row, order=order, backend=backend)
-            row_indexes = list(np.flatnonzero(row))
+            row_indexes = (
+                backend.np.nonzero(row).flatten()
+                if backend.__class__.__name__ == "PyTorchBackend"
+                else list(np.flatnonzero(row))
```

Comment:
maybe adding to the torch backend the `flatnonzero` function could simplify this, namely in the `__init__` defining:
``` python
setattr(self.np, "flatnonzero", lambda x: self.np.nonzero(x).flatten())
```